### ISSUE TITLE
HHVM authentication error

### ISSUE THREAD

**Issue by sianios** [rtCamp/easyengine#492]
I installed ee v3.1.0 on a fresh Vagrant Ubuntu 14.04 environment the installation process was ok but after running the command ```sudo ee stack install --hhvm``` I got the error ```Unable to run apt-get install```. There is a problem with the authentication of the package.
Bellow is the last lines of ```/var/log/ee/ee.log```.

```
W: GPG error: http://dl.hhvm.com trusty InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 5A16E7281BE7A449
2015-04-16 13:25:10,773 (INFO) ee : ESC[94mInstalling packages, please wait ...ESC[0m
Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  ghostscript gsfonts imagemagick-common libboost-context1.54.0
  libboost-filesystem1.54.0 libboost-program-options1.54.0
  libboost-regex1.54.0 libboost-system1.54.0 libboost-thread1.54.0
  libc-client2007e libcupsfilters1 libcupsimage2 libfftw3-double3 libgd3
  libgflags2 libgoogle-glog0 libgs9 libgs9-common libijs-0.35 libjbig2dec0
  libjemalloc1 liblcms2-2 liblqr-1-0 libluajit-5.1-2 libluajit-5.1-common
  libmagickcore5 libmagickwand5 libmcrypt4 libmemcached10 libmysqlclient18
  libonig2 libpaper-utils libpaper1 libtbb2 libunwind8 libvpx1 libxslt1.1
  mlock mysql-common poppler-data
Suggested packages:
  ghostscript-x hpijs uw-mailutils libfftw3-bin libfftw3-dev libgd-tools
  fonts-droid liblcms2-utils libmagickcore5-extra libmcrypt-dev mcrypt
  fcgiwrap nginx-doc ssl-cert poppler-utils fonts-japanese-mincho
  fonts-ipafont-mincho fonts-japanese-gothic fonts-ipafont-gothic
  fonts-arphic-ukai fonts-arphic-uming fonts-unfonts-core
The following NEW packages will be installed:
  ghostscript gsfonts hhvm imagemagick-common libboost-context1.54.0
  libboost-filesystem1.54.0 libboost-program-options1.54.0
  libboost-regex1.54.0 libboost-system1.54.0 libboost-thread1.54.0
  libc-client2007e libcupsfilters1 libcupsimage2 libfftw3-double3 libgd3
  libgflags2 libgoogle-glog0 libgs9 libgs9-common libijs-0.35 libjbig2dec0
  libjemalloc1 liblcms2-2 liblqr-1-0 libluajit-5.1-2 libluajit-5.1-common
  libmagickcore5 libmagickwand5 libmcrypt4 libmemcached10 libmysqlclient18
  libonig2 libpaper-utils libpaper1 libtbb2 libunwind8 libvpx1 libxslt1.1
  mlock mysql-common nginx-common nginx-custom poppler-data
0 upgraded, 43 newly installed, 0 to remove and 24 not upgraded.
Need to get 34.6 MB of archives.
After this operation, 72.9 MB of additional disk space will be used.
WARNING: The following packages cannot be authenticated!
  hhvm
E: There are problems and -y was used without --force-yes
2015-04-16 13:25:11,039 (ERROR) ee : ESC[91mUnable to run apt-get installESC[0m
```

**Comment by gau1991** [comment]
Hi @sianios ,
Can you please try once again to run `ee stack install --hhvm`??
Some time this issue is temporary, and is because of network error, as its unable to get HHVM GPG key.

**Comment by sianios** [comment]
This time it worked fine. Thank you
But I get another error on reloading nginx
```
vagrant@vagrant-ubuntu-trusty-64:~$ sudo ee stack install --hhvm 
Adding repository for HHVM, please wait ...
Updating apt-cache, please wait ...
Installing packages, please wait ...
Restart : hhvm      [OK]
Reload : nginx     [Failed]
[Errno 2] No such file or directory: '/etc/nginx/common/php-hhvm.conf'
```

**Comment by gau1991** [comment]
@sianios ,
I am not able to reproduce the issue. Can you please give output of following command?
```
tail -n100 /var/log/ee/ee.log
```

**Comment by sianios** [comment]
@gau1991 there are the last 100 lines
if you need more here is the complete ee.log http://pastebin.com/De4By2c2
```
Preparing to unpack .../libtbb2_4.2~20130725-1.1ubuntu1_amd64.deb ...
Unpacking libtbb2 (4.2~20130725-1.1ubuntu1) ...
Selecting previously unselected package hhvm.
Preparing to unpack .../hhvm_3.6.1~trusty_amd64.deb ...
Unpacking hhvm (3.6.1~trusty) ...
Selecting previously unselected package libpaper-utils.
Preparing to unpack .../libpaper-utils_1.1.24+nmu2ubuntu3_amd64.deb ...
Unpacking libpaper-utils (1.1.24+nmu2ubuntu3) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Processing triggers for ureadahead (0.100.0-16) ...
Setting up imagemagick-common (8:6.7.7.10-6ubuntu3) ...
Setting up libavahi-common-data:amd64 (0.6.31-4ubuntu1) ...
Setting up libavahi-common3:amd64 (0.6.31-4ubuntu1) ...
Setting up libavahi-client3:amd64 (0.6.31-4ubuntu1) ...
Setting up libboost-context1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libboost-system1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libboost-filesystem1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libboost-program-options1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libboost-regex1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libboost-thread1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libcups2:amd64 (1.7.2-0ubuntu1.5) ...
Setting up libjpeg-turbo8:amd64 (1.3.0-0ubuntu2) ...
Setting up libjpeg8:amd64 (8c-2ubuntu8) ...
Setting up libjbig0:amd64 (2.0-2ubuntu4.1) ...
Setting up libtiff5:amd64 (4.0.3-7ubuntu0.3) ...
Setting up libcupsfilters1:amd64 (1.0.52-0ubuntu1.4) ...
Setting up libcupsimage2:amd64 (1.7.2-0ubuntu1.5) ...
Setting up libfftw3-double3:amd64 (3.3.3-7ubuntu3) ...
Setting up fonts-dejavu-core (2.34-1ubuntu1) ...
Setting up fontconfig-config (2.11.0-0ubuntu4.1) ...
Setting up libfontconfig1:amd64 (2.11.0-0ubuntu4.1) ...
Setting up libvpx1:amd64 (1.3.0-2) ...
Setting up libgd3:amd64 (2.1.0-3) ...
Setting up libgflags2 (2.0-1.1ubuntu1) ...
Setting up libunwind8 (1.1-2.2ubuntu3) ...
Setting up libgoogle-glog0 (0.3.3-1) ...
Setting up libjasper1:amd64 (1.900.1-14ubuntu3.2) ...
Setting up liblcms2-2:amd64 (2.5-0ubuntu4) ...
Setting up liblqr-1-0:amd64 (0.4.1-2ubuntu1) ...
Setting up libltdl7:amd64 (2.4.2-1.7ubuntu1) ...
Setting up libmagickcore5:amd64 (8:6.7.7.10-6ubuntu3) ...
Setting up libmagickwand5:amd64 (8:6.7.7.10-6ubuntu3) ...
Setting up libmemcached10:amd64 (1.0.8-1ubuntu2) ...
Setting up mysql-common (5.5.41-0ubuntu0.14.04.1) ...
Setting up libmysqlclient18:amd64 (5.5.41-0ubuntu0.14.04.1) ...
Setting up libpaper1:amd64 (1.1.24+nmu2ubuntu3) ...

Creating config file /etc/papersize with new version
Setting up libxslt1.1:amd64 (1.1.28-2build1) ...
Setting up poppler-data (0.4.6-4) ...
Setting up libjemalloc1 (3.5.1-2) ...
Setting up libijs-0.35 (0.35-8build1) ...
Setting up libjbig2dec0 (0.11+20120125-1ubuntu1) ...
Setting up libgs9-common (9.10~dfsg-0ubuntu10.2) ...
update-alternatives: using /usr/share/ghostscript/9.10 to provide /usr/share/ghostscript/current (ghostscript-current) in auto mode
Setting up libgs9 (9.10~dfsg-0ubuntu10.2) ...
Setting up gsfonts (1:8.11+urwcyr1.0.7~pre44-4.2ubuntu1) ...
Setting up ghostscript (9.10~dfsg-0ubuntu10.2) ...
Setting up mlock (8:2007f~dfsg-2) ...
Setting up libc-client2007e (8:2007f~dfsg-2) ...
Setting up libmcrypt4 (2.5.8-3.1ubuntu1) ...
Setting up libonig2 (5.9.1-1ubuntu1) ...
Setting up libtbb2 (4.2~20130725-1.1ubuntu1) ...
Setting up hhvm (3.6.1~trusty) ...
update-alternatives: using /usr/bin/hhvm to provide /usr/bin/php (php) in auto mode
********************************************************************
* HHVM is installed.
* 
* Running PHP web scripts with HHVM is done by having your webserver talk to HHVM
* over FastCGI. Install nginx or Apache, and then:
* $ sudo /usr/share/hhvm/install_fastcgi.sh
* $ sudo /etc/init.d/hhvm restart
* (if using nginx)  $ sudo /etc/init.d/nginx restart
* (if using apache) $ sudo /etc/init.d/apache restart
* 
* Detailed FastCGI directions are online at:
* https://github.com/facebook/hhvm/wiki/FastCGI
* 
* If you're using HHVM to run web scripts, you probably want it to start at boot:
* $ sudo update-rc.d hhvm defaults
* 
* Running command-line scripts with HHVM requires no special setup:
* $ hhvm whatever.php
* 
* You can use HHVM for /usr/bin/php even if you have php-cli installed:
* $ sudo /usr/bin/update-alternatives --install /usr/bin/php php /usr/bin/hhvm 60
********************************************************************
Setting up libpaper-utils (1.1.24+nmu2ubuntu3) ...
Processing triggers for libc-bin (2.19-0ubuntu6.6) ...
Processing triggers for ureadahead (0.100.0-16) ...
2015-04-16 13:59:27,381 (DEBUG) ee : [95mCalling post_pref[0m
2015-04-16 13:59:27,381 (DEBUG) ee : [95mDoning search and replace, File:/etc/hhvm/server.ini,Source string:9000, Dest String:8000[0m
2015-04-16 13:59:27,381 (DEBUG) ee : [95mDoning search and replace, File:/etc/nginx/hhvm.conf,Source string:9000, Dest String:8000[0m
2015-04-16 13:59:27,382 (DEBUG) ee : [95mEEGit: git init at /etc/hhvm[0m
2015-04-16 13:59:27,410 (DEBUG) ee : [95mEEGit: git commit at /etc/hhvm[0m
2015-04-16 13:59:27,436 (INFO) ee : [94mRestart : hhvm      [0m
2015-04-16 13:59:37,505 (INFO) ee : [94m[[0mOK[94m][0m
2015-04-16 13:59:37,505 (INFO) ee : [94mReload : nginx     [0m
2015-04-16 13:59:37,507 (DEBUG) ee : [95m/bin/sh: 1: nginx: not found[0m
2015-04-16 13:59:37,507 (INFO) ee : [94m[[91mFailed[94m][0m
2015-04-16 13:59:37,508 (DEBUG) ee : [95mWritting the nginx configuration to file /etc/nginx/common/php-hhvm.conf[0m
```

**Comment by gau1991** [comment]
Hi, @sianios ,
Thanks for the response ,
Here HHVM is trying to setup its Nginx config, but in your case Nginx is not installed so its failing.
Right now temporary fix for your issue is to install Nginx also.

We are working on this issue, we will fix and release a new EasyEngine soon.

**Comment by gau1991** [comment]
Fixed in EasyEngine 3.1.1
